lib/sidekiq_alive/redis/redis_client_gem.rb

```diff
@@ -7,32 +7,45 @@ module Redis
     # Wrapper for `redis-client` gem used by `sidekiq` > 7
     # https://github.com/redis-rb/redis-client
     class RedisClientGem < Base
+      def initialize
+        super
+
+        @capsule = Sidekiq.default_configuration.capsules[CAPSULE_NAME]
+      end
+
       def set(key, time:, ex:)
-        Sidekiq.redis { |redis| redis.call("SET", key, time, ex: ex) }
+        redis { |r| r.call("SET", key, time, ex: ex) }
       end
 
       def get(key)
-        Sidekiq.redis { |redis| redis.call("GET", key) }
+        redis { |r| r.call("GET", key) }
       end
 
       def zadd(set_key, ex, key)
-        Sidekiq.redis { |redis| redis.call("ZADD", set_key, ex, key) }
+        redis { |r| r.call("ZADD", set_key, ex, key) }
       end
 
       def zrange(set_key, start, stop)
-        Sidekiq.redis { |redis| redis.call("ZRANGE", set_key, start, stop) }
+        redis { |r| r.call("ZRANGE", set_key, start, stop) }
       end
 
       def zrangebyscore(set_key, min, max)
-        Sidekiq.redis { |redis| redis.call("ZRANGEBYSCORE", set_key, min, max) }
+        redis { |r| r.call("ZRANGEBYSCORE", set_key, min, max) }
       end
 
       def zrem(set_key, key)
-        Sidekiq.redis { |redis| redis.call("ZREM", set_key, key) }
+        redis { |r| r.call("ZREM", set_key, key) }
       end
 
       def delete(key)
-        Sidekiq.redis { |redis| redis.call("DEL", key) }
+        redis { |r| r.call("DEL", key) }
+      end
+
+      private
+
+      def redis(&block)
+        # Default to Sidekiq.redis if capsule is not configured yet but redis adapter is accessed
+        (@capsule || Sidekiq).redis(&block)
```

Comment:
Sidekiq documentation states that capsule specific redis pool should be used. We need this fallback because method returning redis adapter is a class method and can be accessed before actual configuration has been performed (it is used in tests like that currently).